drltrace/drltrace.cpp

```diff
@@ -414,58 +408,23 @@ event_exit(void)
 static void
 options_init(client_id_t id)
 {
-    const char *opstr = dr_get_options(id);
-    const char *s;
-    char token[OPTION_MAX_LENGTH];
-    /* default values */
-    dr_snprintf(options.logdir, BUFFER_SIZE_ELEMENTS(options.logdir), "-");
-    options.max_args = 6;
-    options.num_unknown_args = 2;
+    std::string parse_err;
+    if (!dr_parse_options(id, &parse_err, NULL))
+        ASSERT(false, "unable to parse options specified for drltracelib");
+
     op_print_stderr = true;
-    /* XXX: we have to use droption here instead of manual parsing */
-    for (s = dr_get_token(opstr, token, BUFFER_SIZE_ELEMENTS(token));
-         s != NULL;
-         s = dr_get_token(s, token, BUFFER_SIZE_ELEMENTS(token))) {
-        if (strcmp(token, "-logdir") == 0) {
-            s = dr_get_token(s, options.logdir,
-                             BUFFER_SIZE_ELEMENTS(options.logdir));
-            USAGE_CHECK(s != NULL, "missing logdir path");
-        } else if (strcmp(token, "-only_from_app") == 0) {
-            options.only_from_app = true;
-        } else if (strcmp(token, "-only_to_lib") == 0) {
-            s = dr_get_token(s, options.only_to_lib,
-                             BUFFER_SIZE_ELEMENTS(options.only_to_lib));
-            USAGE_CHECK(s != NULL, "missing library name");
-        } else if (strcmp(token, "-ignore_underscore") == 0) {
-            options.ignore_underscore = true;
-        } else if (strcmp(token, "-num_unknown_args") == 0) {
-            s = dr_get_token(s, token, BUFFER_SIZE_ELEMENTS(token));
-            USAGE_CHECK(s != NULL, "missing -num_unknown_args number");
-            if (s != NULL) {
-                int res = dr_sscanf(token, "%u", &options.num_unknown_args);
-                USAGE_CHECK(res == 1, "invalid -num_unknown_args number");
-            }
-        } else if (strcmp(token, "-num_max_args") == 0) {
-            s = dr_get_token(s, token, BUFFER_SIZE_ELEMENTS(token));
-            USAGE_CHECK(s != NULL, "missing -num_max_args number");
-            if (s != NULL) {
-                int res = dr_sscanf(token, "%u", &options.max_args);
-                USAGE_CHECK(res == 1, "invalid -num_max_args number");
-            }
-        } else if (strcmp(token, "-verbose") == 0) {
-            s = dr_get_token(s, token, BUFFER_SIZE_ELEMENTS(token));
-            USAGE_CHECK(s != NULL, "missing -verbose number");
-            if (s != NULL) {
-                int res = dr_sscanf(token, "%u", &verbose);
-                USAGE_CHECK(res == 1, "invalid -verbose number");
-            }
-        } else if (strcmp(token, "-print_ret_addr") == 0)
-            options.print_ret_addr = true;
-        else {
-            NOTIFY("UNRECOGNIZED OPTION: \"%s\""NL, token);
-            USAGE_CHECK(false, "invalid option");
-        }
-    }
+
+    /* to avoid numerous get_value() calls we copy all options here */
```

Comment:
I wanted to save a little bit time at runtime (especially in case of print_ret_addr and etc.). Do you think it is negligible, right ?
Anyway, fixed everything.
Done.